### PR TITLE
GH-33

### PR DESCRIPTION
**Pull request by akila-sama**

### CODE REVIEW

File: classes/class-settings.php [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+	/**
+	 * Add Settings Page
+	 *
+	 * Adds the settings page to the WordPress admin menu.
+	 *
+	 * @since 1.0.0
+	 */
+	public function add_settings_page() {
+		add_menu_page(
```
Comment (nilesh-32236) [inline classes/class-settings.php]: Do not create separate menu.
Add this menu in plugin menu using add_submenu_page.

File: classes/class-settings.php [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+	 * @since 1.0.0
+	 */
+	public function akila_settings_page() {
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return;
+		}
+
+		if ( isset( $_POST['akila_submit'] ) && wp_verify_nonce( $_POST['akila_settings_nonce'], 'akila_settings_action' ) ) {
+			$email_notifications = isset( $_POST['akila_email_notifications'] ) ? 1 : 0;
+			update_option( 'akila_email_notifications', $email_notifications );
```
Comment (nilesh-32236) [inline classes/class-settings.php]: Store options in one array.
Do not store same page options in deference option.

File: classes/class-settings.php [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+			?>
+			<div class="updated"><p><?php esc_html_e( 'Settings saved successfully.', 'akila-portfolio' ); ?></p></div>
+			<?php
+		}
+
+		$email_notifications    = get_option( 'akila_email_notifications', 1 );
+		$notification_frequency = get_option( 'akila_notification_frequency', 'daily' );
+		//Searching (akila_notification_frequency) in the database will show that daily, monthly, weekly are set.
+		?>
+		<div class="wrap">
```
Comment (nilesh-32236) [inline classes/class-settings.php]: Create separate template for html code.

File: classes/class-settings.php [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+	 * to enable or disable email notifications for the Akila Portfolio plugin.
+	 *
+	 * @since 1.0.0
+	 */
+	public function akila_settings_page() {
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return;
+		}
+
+		if ( isset( $_POST['akila_submit'] ) && wp_verify_nonce( $_POST['akila_settings_nonce'], 'akila_settings_action' ) ) {
```
Comment (nilesh-32236) [inline classes/class-settings.php]: use ajax request to store data.

File: classes/class-cron.php [hunk 5]
```diff
@@ -25,16 +25,20 @@ public function __construct() {
...
 		}
 	}
 
+
 	/**
 	 * Send email notifications.
 	 *
-	 * Retrieves all portfolio posts, extracts email addresses from post meta,
-	 * and sends daily notification emails.
+	 * Retrieves all published portfolio posts and sends an email notification
```
Comment (uvesh-bhatti) [inline classes/class-cron.php:46]: add proper documantation 
use @ since @ return

File: classes/class-cron.php [hunk 6]
```diff
@@ -25,16 +25,20 @@ public function __construct() {
 	 * Schedules the event to send email notifications daily if not already scheduled.
 	 */
 	public function schedule_cron_event() {
-		if ( ! wp_next_scheduled( 'akila_portfolio_send_email_notifications' ) ) {
+		// Check if email notifications are enabled
+		$options = get_option( 'akila_portfolio_notification_options', array() );
+		// Schedule cron event only if email notifications are enabled
+		if ( $options['email_notifications'] && ! wp_next_scheduled( 'akila_portfolio_send_email_notifications' ) ) {
```
Comment (uvesh-bhatti) [inline classes/class-cron.php]: this give me warning in error log Please fix it
Warning: Undefined array key "email_notifications"

File: classes/class-endpoints.php [hunk 7]
```diff
@@ -19,27 +20,51 @@ public function __construct() {
 	 */
 	public function ak_register_custom_endpoints() {
 		register_rest_route(
-			'v1',
-			'/custom-endpoint/',
+			'akila-portfolio/v1', // Namespace and version for your endpoint
+			'/portfolio-posts', // Endpoint path
+			//http://localhost/wp_plugin_dev/wp-json/akila-portfolio/v1/portfolio-posts
```
Comment (uvesh-bhatti) [inline classes/class-endpoints.php]: if commented code is unnecessary so remove it

File: classes/class-pluginpage.php [hunk 8]
```diff
@@ -17,61 +20,17 @@ class PluginPage {
 
 	public function __construct() {
 		add_action( 'admin_menu', array( $this, 'ak_custom_menu' ) );
-		add_action( 'admin_enqueue_scripts', array( $this, 'ak_enqueue_my_plugin_ajax_script' ) );
+		add_action( 'admin_enqueue_scripts', array( $this, 'admin_enqueue_scripts' ) );
```
Comment (uvesh-bhatti) [inline classes/class-pluginpage.php]: do not use same name as hook name use as relevant name of callback fun

File: classes/class-settings.php [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+
+namespace APortfolio;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Class Settings
+ * Handles settings page functionalities for the plugin.
```
Comment (uvesh-bhatti) [inline classes/class-settings.php:11]: add proper documantation
use @ since @ return

File: classes/class-pluginpage.php [hunk 10]
```diff
@@ -17,61 +20,17 @@ class PluginPage {
...
-			wp_localize_script(
-				'akila-portfolio-js',
-				'ak_my_plugin',
-				array(
-					'ajax_url' => admin_url( 'admin-ajax.php' ),
-				)
-			);
-		}
-	}
+		add_action( 'admin_menu', array( $this, 'ak_add_settings_page_submenu' ) );
```
Comment (uvesh-bhatti) [inline classes/class-pluginpage.php]: this hook is already define in class-setting.php so remove it.

File: classes/class-settings.php [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+		wp_send_json_success( array( 'message' => esc_html__( 'Settings saved successfully.', 'akila-portfolio' ) ) );
+	}
+
+	/**
+	 * Get notification interval based on selected frequency.
+	 *
+	 * @param string $frequency Notification frequency.
+	 * @return string Interval for scheduling cron job.
+	 */
+	private function ak_get_notification_interval( $frequency ) {
```
Comment (uvesh-bhatti) [inline classes/class-settings.php]: if you don't use this function so please remove it.

File: classes/class-pluginpage.php [hunk 12]
```diff
@@ -17,61 +20,17 @@ class PluginPage {
...
+		add_action( 'wp_ajax_save_settings', array( $this, 'ak_save_settings' ) );
```
Comment (uvesh-bhatti) [inline classes/class-pluginpage.php]: this ajax request is not work because it callback is not define

File: classes/class-submenu.php [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+namespace APortfolio;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Class Submenu
+ * Handles submenu functionalities for the plugin.
+ */
```
Comment (uvesh-bhatti) [inline classes/class-submenu.php:13]: add proper documantation
use @ since @ return

File: classes/class-submenu.php [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+class Submenu {
+
+	public function __construct() {
+		add_action( 'admin_menu', array( $this, 'ak_custom_submenu' ) );
+	}
+
+	/**
+	 * Add a submenu page.
+	 * Registers a submenu page for managing REST API functionalities.
```
Comment (uvesh-bhatti) [inline classes/class-submenu.php:22]: add proper documantation
use @ since @ return etc

File: classes/class-submenu.php [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+			'manage_options',
+			'ak_custom-submenu-slug', // Menu slug
+			array( $this, 'ak_display_submenu_details' )
+		);
+	}
+
+	/**
+	 * Update the submenu page callback function to display portfolio posts.
+	 * Callback function to display portfolio posts in the submenu page.
+	 */
```
Comment (uvesh-bhatti) [inline classes/class-submenu.php:40]: add proper documantation
use @ since @ return etc

File: js/akila-portfolio.js [hunk 16]
```diff
@@ -211,3 +213,34 @@ jQuery(document).ready(function ($) {
 		});
 	});
 });
+
+
+//email setting page
```
Comment (uvesh-bhatti) [inline js/akila-portfolio.js]: add proper documantation
use @ since @ returnetc

File: js/akila-portfolio.js [hunk 17]
```diff
@@ -211,3 +213,34 @@ jQuery(document).ready(function ($) {
 		});
 	});
 });
+
+
+//email setting page 
+jQuery(document).ready(function($) {
```
Comment (uvesh-bhatti) [inline js/akila-portfolio.js]: use coding standard for all js

File: templates/tmpl-portfolio-post.php [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+				<td>{{ item.email }}</td>
+				<td>{{ item.phone }}</td>
+				<td>{{ item.address }}</td>
+				<td>{{ item.date }}</td>
+				<td><button class="delete-portfolio-post" data-post-id="{{ item.post_id }}">{{ '<?php esc_html_e( 'Delete', 'akila-portfolio' ); ?>' }}</button></td>
+			</tr>
+		<# }) #>
+	</tbody>
+</table>
+</script>
```
Comment (uvesh-bhatti) [inline templates/tmpl-portfolio-post.php]: File must end with a newline

**Pull request merged**